app/javascript/packs/cms.ts

```diff
@@ -332,14 +312,25 @@ function setUpDropdownButtonClose () {
 }
 
 function closeAllDropdowns (exception?: HTMLElement) {
-  document.querySelectorAll('.dropdown.expanded')
+  document.querySelectorAll('.pf-c-dropdown.pf-m-expanded')
     .forEach(dropdown => {
-      const buttonGroup = dropdown.parentElement as HTMLDivElement
-
-      if (exception && buttonGroup.contains(exception)) {
+      if (exception && dropdown.contains(exception)) {
         return
       }
-
-      dropdown.classList.remove('expanded')
+      dropdown.classList.remove('pf-m-expanded')
+      dropdown.querySelector("ul")!.style.display = 'none'
     })
 }
+
+/**
+ * Set up CMS Preview button to act as a link
+ */
+function setUpPreviewButton () {
```

Comment:
The purpose of this function is to open a URL stored in the `data-url` attribute of the `<button>` element named "Preview". In the previous implementation it was an `<a>` element, so it was not needed. However, in PF4 dropdown it should be a button.

I could have just added an `onclick` handler to the button (like I did for the "New X" button), but in this case the URL is not static, and can get updated if the path of the template is changed, see `app/views/provider/admin/cms/templates/update.js.erb`.